### ISSUE TITLE
Implement Connection Service File

### ISSUE THREAD

**Issue by feikesteenbergen** [lib/pq#538]
Currently the driver does not support specifying a connection using a service description.
Details: https://www.postgresql.org/docs/current/static/libpq-pgservice.html

I'm currently not a go programmer myself, however looking at the code I might be able to create a proof-of-concept implementation.

Basically, what needs to happen that iff a service is specified, we need to read the libpq parameters from the Connection Service File. For a start, we could use the same logic (and limitations) as the pgpass reading: https://github.com/lib/pq/blob/master/conn.go#L159-L168

Questions:
- Does anyone - other than me - think this is useful?
- The Connection Service File is in an ini format, should we use an external ini (or toml) library to implement this?

**Comment by cuboci** [comment]
I would love this. I've started using a ~/.pg_service.conf file for cleaner psql command line parameters and would be quite happy if I could use the same service=foo parameter for my PostgreSQL-using go programs.

**Comment by dskoda1** [comment]
@feikesteenbergen @cuboci I also agree this would be useful. It would help bridge the gap between Postgres clients in other languages like [`psycopg2`](https://github.com/psycopg/psycopg2) which implements this.

**Comment by jl-** [comment]
Any update on this feature?

**Comment by schoukri** [comment]
I too would love to see this feature implemented. I use a command line tool called [migrate](https://github.com/golang-migrate/migrate), which is written in go and uses lib/pq. I was surprised to learn that it didn't support the service file (especially given that it does support the pgpass file).

**Comment by ivanov17** [comment]
Is this still planned?
